protocol/linkpreview/linkpreview_test.go

```diff
@@ -88,106 +155,130 @@ func TestGetLinks(t *testing.T) {
-				Description: "GitHub is where over 100 million developers shape the future of software, together. Contribute to the open source community, manage your Git repositories, review code like a pro, track bugs and fea...",
-				Hostname:    "github.com",
-				Title:       "GitHub: Let’s build from here",
-				URL:         "https://github.com/",
-				Thumbnail: common.LinkPreviewThumbnail{
-					Width:   1200,
-					Height:  630,
-					URL:     "",
-					DataURI: "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAABLAAAAJ2CAMAAAB4",
-				},
-			},
-		},
-		{
-			url: "https://github.com/status-im/status-mobile/issues/15469",
-			expected: common.LinkPreview{
-				Description: "Designs https://www.figma.com/file/wA8Epdki2OWa8Vr067PCNQ/Composer-for-Mobile?node-id=2102-232933&t=tTYKjMpICnzwF5Zv-0 Out of scope Enable link previews (we can assume for now that is always on) Mu...",
-				Hostname:    "github.com",
-				Title:       "Allow users to customize links · Issue #15469 · status-im/status-mobile",
-				URL:         "https://github.com/status-im/status-mobile/issues/15469",
-				Thumbnail: common.LinkPreviewThumbnail{
-					Width:   1200,
-					Height:  600,
-					URL:     "",
-					DataURI: "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAABLAAAAJYCAYAAABy",
-				},
-			},
-		},
-		{
-			url: "https://www.imdb.com/title/tt0117500/",
-			expected: common.LinkPreview{
-				Description: "The Rock: Directed by Michael Bay. With Sean Connery, Nicolas Cage, Ed Harris, John Spencer. A mild-mannered chemist and an ex-con must lead the counterstrike when a rogue group of military men, led by a renegade general, threaten a nerve gas attack from Alcatraz against San Francisco.",
-				Hostname:    "www.imdb.com",
-				Title:       "The Rock (1996) - IMDb",
-				URL:         "https://www.imdb.com/title/tt0117500/",
-				Thumbnail: common.LinkPreviewThumbnail{
-					Width:   1000,
-					Height:  1481,
-					URL:     "",
-					DataURI: "data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wCEAAkJCgg",
-				},
-			},
-		},
-		{
-			url: "https://www.youtube.com/watch?v=lE4UXdJSJM4",
-			expected: common.LinkPreview{
-				URL:         "https://www.youtube.com/watch?v=lE4UXdJSJM4",
-				Hostname:    "www.youtube.com",
-				Title:       "Interview with a GNU/Linux user - Partition 1",
-				Description: "GNU/Linux Operating SystemInterview with a GNU/Linux user with Richie Guix - aired on © The GNU Linux.Programmer humorLinux humorProgramming jokesProgramming...",
-				Thumbnail: common.LinkPreviewThumbnail{
-					Width:   1280,
-					Height:  720,
-					DataURI: "data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wCEAAUDBA8",
-				},
-			},
+func readAsset(t *testing.T, filename string) []byte {
+	b, err := ioutil.ReadFile("../../_assets/tests/" + filename)
+	require.NoError(t, err)
+	return b
+}
+
+func Test_UnfurlURLs_YouTube(t *testing.T) {
+	url := "https://www.youtube.com/watch?v=lE4UXdJSJM4"
+	thumbnailURL := "https://i.ytimg.com/vi/lE4UXdJSJM4/maxresdefault.jpg"
```

Comment:
Do these tests need to consider segregating getting data from parsing data?

See https://github.com/status-im/status-go/issues/3529

Reply:
Good question @Samyoul, but I'm inferring a little bit what you mean, please correct me if anything is off. So far, I've focused on testing only the public function `UnfurlURLs`, but because it does basically everything under the hood, its tests are more sociable than solitary.

I think I could do a better job of testing some of the private functions too, since they have a narrower responsibility, and it would be easier to test this segregation of data fetching vs parsing you mention.

Reply:
Slightly off topic, I saw many examples in the code about using a mock server, but in retrospect, I don't remember anymore why I didn't go that route, which in my experience (in other languages) works really well too. In the end, I opted for direct control over a RoundTripper to give me more control (?). Maybe overkill? I must say I've had good results with both approaches.

And your comments on the issue about the flaky tests are on point :)

Reply:
You know, I really have enjoyed reviewing this PR. Testing http requests are always tricky because the http client interface is generally embedded and segregating the getting concerns from the parsing concerns isn't a focus.

A mock client request verses a mock server response is basically removing the need for outbound calls. I've always focused on making the server mockable so that the client behaves as normally as possible during tests. But in your case I think that the server is not required because you pass in (inject) the client dependency and so you only really test the data parsing concern of your functionality.

Reply:
Actually I may have just become a convert to client centric testing.